lib/cli/src/commands/run/wasi.rs

```diff
@@ -116,6 +146,47 @@ pub struct Wasi {
     pub deny_multiple_wasi_versions: bool,
 }
 
+/// Various triggers that will cause the runtime to take snapshot
+/// of the WASM state and store it in the snapshot file.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SnapshotTrigger {
+    /// Triggered when all the threads in the process goes idle
+    Idle,
+    /// Triggered when a listen syscall is invoked on a socket
+    Listen,
+    /// Triggered when the process reads stdin for the first time
+    Stdin,
+    /// Triggered periodically (default 10 seconds) which can be specified using the `snapshot-period` option
+    Periodic,
```

Comment:
I'm a bit confused by this. On the snapshot-period CLI argument it puts is as a delay between process -> dump data. But here it says to be a time period repeated over time (which I don't understand)

Reply:
Renamed this from periodic to timer to make it more clear what its is
https://github.com/wasmerio/wasmer/pull/4263/commits/8f11ac78b00d3f07b774e776d20e54221dc9e33f